packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -245,6 +255,42 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
         startBlock_ = Storage.getUint(START_BLOCK_SLOT);
     }
 
+    /// @notice Getter for the gas paying asset address.
+    function gasPayingToken() external view returns (address addr_, uint8 decimals_) {
+        (addr_, decimals_) = GasPayingToken.getToken();
+    }
+
+    /// @notice Getter for the gas paying token name.
+    function gasPayingTokenName() external view returns (string memory name_) {
+        name_ = GasPayingToken.getName();
+    }
+
+    /// @notice Getter for the gas paying token symbol.
+    function gasPayingTokenSymbol() external view returns (string memory symbol_) {
+        symbol_ = GasPayingToken.getSymbol();
+    }
+
+    /// @notice Internal setter for the gas paying token address, includes validation.
+    /// @param _token Address of the gas paying token.
+    function _setGasPayingToken(address _token) internal {
+        if (_token != address(0) && _token != Constants.ETHER) {
```

Comment:
We could add another condition to this branch for if the address read from state is not set to anything